src/details/ArborX_DetailsPermutedData.hpp

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_PERMUTED_DATA_HPP
+#define ARBORX_DETAILS_PERMUTED_DATA_HPP
+
+#include <ArborX_AccessTraits.hpp>
+
+namespace ArborX
+{
+
+namespace Details
+{
+
+template <typename Data, typename Permute, bool AttachIndices = false>
+struct PermutedData
+{
+  Data _data;
+  Permute _permute;
+  KOKKOS_FUNCTION auto &operator()(int i) const { return _data(_permute(i)); }
```

Comment:
When would you want the indirection here but not want to attach the indices?

Reply:
I don't understand the question.